### DISCUSSION TITLE
Building dynamic form based on API response and populate data

### DISCUSSION THREAD

**Discussion by anushkamds** [react-hook-form/react-hook-form#9256] [Q&A]
Hi guys, 
I am trying to build a dynamic from populate data for that form. I would be getting object as follows with values and I will be using checkboxes. 
```json 
{
"admin": true,
"manager": false,
"client": true
}
```
these values are dynamic and can more than this. 
After I loaded the data to the component, I need to render checkboxes with values(checked or not). I tried with `reset` to set values to the form and didn't work. I guess that because I haven set dynamic field to `useForm` function which I can't do when component is loaded.  
Is there any way to build dynamic form with API response ?
Thank you in advance

**Comment by leapful** [comment]
@anushkamds : Could you please share a codesandbox url to re-produce your issue ?
